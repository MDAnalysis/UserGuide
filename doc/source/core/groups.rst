.. -*- coding: utf-8 -*-

====================
Groups
====================


For convenience, chemically meaningful groups of Atoms such as a 
Residue or a Segment (typically a whole molecule or all of the solvent) 
also exist as containers, as well as groups of these units (
ResidueGroup, SegmentGroup).

AtomGroups
==========
AtomGroup instances can be easily created 
(e.g., from an AtomGroup.select_atoms() selection or simply by slicing).

