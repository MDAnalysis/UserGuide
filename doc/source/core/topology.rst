.. -*- coding: utf-8 -*-

====================
Topology
====================

MDA Topology =/= 'force field' topology. Connectivity only. See ParmEd etc for other stuff.

Adding attributes to Topology
=============================

stuff