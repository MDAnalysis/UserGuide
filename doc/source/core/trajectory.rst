.. -*- coding: utf-8 -*-

====================
Trajectory
====================

Not intuitive.

.positions() updates

Auxiliary trajectory readers
============================

