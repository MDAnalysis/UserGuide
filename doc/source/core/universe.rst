.. -*- coding: utf-8 -*-

====================
Universes
====================

| *If you wish to make an apple pie from scratch, you must first invent the universe.*
| - Carl Sagan, Cosmos

| *In the beginning the Universe was created.*
| *This has made a lot of people very angry and been widely regarded as a bad move.*
| â€• Douglas Adams, The Restaurant at the End of the Universe

The Universe class ties a topology and a trajectory together. 
Almost all code in MDAnalysis starts with a Universe.

Creating a Universe
===================

Loading from files
------------------

.. code-block:: python

u = Universe(topology, trajectory)          # read system from file(s)
u = Universe(pdbfile)                       # read atoms and coordinates from PDB or GRO
u = Universe(topology, [traj1, traj2, ...]) # read from a list of trajectories
u = Universe(topology, traj1, traj2, ...)   # read from multiple trajectories

Many words here

Constructing from AtomGroups
----------------------------

Merge()

