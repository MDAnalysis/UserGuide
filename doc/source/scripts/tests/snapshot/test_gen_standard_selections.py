from unittest.mock import patch

from gen_standard_selections import StandardSelectionTable
from MDAnalysis.core import selection as sel


def test_StandardSelectionTable_protein(snapshot):
    with patch("builtins.open"):
        ss = StandardSelectionTable(
            "protein",
            klass=sel.ProteinSelection,
            attribute_name="prot_res",
            sort=True,
        )
    assert ss.table_writer.lines == snapshot


def test_StandardSelectionTable_protein_backbone(snapshot):
    with patch("builtins.open"):
        ss = StandardSelectionTable(
            "protein_backbone",
            klass=sel.BackboneSelection,
            attribute_name="bb_atoms",
            sort=True,
        )
    assert ss.table_writer.lines == snapshot


def test_StandardSelectionTable_nucleic(snapshot):
    with patch("builtins.open"):
        ss = StandardSelectionTable(
            "nucleic",
            klass=sel.NucleicSelection,
            attribute_name="nucl_res",
            sort=True,
        )
    assert ss.table_writer.lines == snapshot


def test_StandardSelectionTable_nucleic_backbone(snapshot):
    with patch("builtins.open"):
        ss = StandardSelectionTable(
            "nucleic_backbone",
            klass=sel.NucleicBackboneSelection,
            attribute_name="bb_atoms",
            sort=True,
        )
    assert ss.table_writer.lines == snapshot


def test_StandardSelectionTable_base(snapshot):
    with patch("builtins.open"):
        ss = StandardSelectionTable(
            "base",
            klass=sel.BaseSelection,
            attribute_name="base_atoms",
            sort=True,
        )
    assert ss.table_writer.lines == snapshot


def test_StandardSelectionTable_nucleic_sugar(snapshot):
    with patch("builtins.open"):
        ss = StandardSelectionTable(
            "nucleic_sugar",
            klass=sel.NucleicSugarSelection,
            attribute_name="sug_atoms",
            sort=True,
        )
    assert ss.table_writer.lines == snapshot
