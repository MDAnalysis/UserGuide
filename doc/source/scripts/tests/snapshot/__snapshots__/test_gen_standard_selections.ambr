# serializer version: 1
# name: test_StandardSelectionTable_base
  list([
    list([
      'C2',
      'C4',
      'C5',
      'C5M',
      'C6',
      'C8',
      'N1',
      'N2',
    ]),
    list([
      'N3',
      'N4',
      'N6',
      'N7',
      'N9',
      'O2',
      'O4',
      'O6',
    ]),
  ])
# ---
# name: test_StandardSelectionTable_nucleic
  list([
    list([
      'A',
      'ADE',
      'C',
      'CYT',
      'DA',
      'DA3',
      'DA5',
      'DC',
    ]),
    list([
      'DC3',
      'DC5',
      'DG',
      'DG3',
      'DG5',
      'DT',
      'DT3',
      'DT5',
    ]),
    list([
      'G',
      'GUA',
      'RA',
      'RA3',
      'RA5',
      'RC',
      'RC3',
      'RC5',
    ]),
    list([
      'RG',
      'RG3',
      'RG5',
      'RU',
      'RU3',
      'RU5',
      'T',
      'THY',
    ]),
    list([
      'U',
      'URA',
    ]),
  ])
# ---
# name: test_StandardSelectionTable_nucleic_backbone
  list([
    list([
      "C3'",
      "C5'",
      "O3'",
      "O5'",
      'P',
    ]),
  ])
# ---
# name: test_StandardSelectionTable_nucleic_sugar
  list([
    list([
      "C1'",
      "C2'",
      "C3'",
      "C4'",
      "O4'",
    ]),
  ])
# ---
# name: test_StandardSelectionTable_protein
  list([
    list([
      'ACE',
      'ALA',
      'ALAD',
      'ARG',
      'ARGN',
      'ASF',
      'ASH',
      'ASN',
    ]),
    list([
      'ASN1',
      'ASP',
      'ASPH',
      'CALA',
      'CARG',
      'CASF',
      'CASN',
      'CASP',
    ]),
    list([
      'CCYS',
      'CCYX',
      'CGLN',
      'CGLU',
      'CGLY',
      'CHID',
      'CHIE',
      'CHIP',
    ]),
    list([
      'CILE',
      'CLEU',
      'CLYS',
      'CME',
      'CMET',
      'CPHE',
      'CPRO',
      'CSER',
    ]),
    list([
      'CTHR',
      'CTRP',
      'CTYR',
      'CVAL',
      'CYM',
      'CYS',
      'CYS1',
      'CYS2',
    ]),
    list([
      'CYSH',
      'CYX',
      'DAB',
      'GLH',
      'GLN',
      'GLU',
      'GLUH',
      'GLY',
    ]),
    list([
      'HID',
      'HIE',
      'HIP',
      'HIS',
      'HIS1',
      'HIS2',
      'HISA',
      'HISB',
    ]),
    list([
      'HISD',
      'HISE',
      'HISH',
      'HSD',
      'HSE',
      'HSP',
      'HYP',
      'ILE',
    ]),
    list([
      'LEU',
      'LYN',
      'LYS',
      'LYSH',
      'MET',
      'MSE',
      'NALA',
      'NARG',
    ]),
    list([
      'NASN',
      'NASP',
      'NCYS',
      'NCYX',
      'NGLN',
      'NGLU',
      'NGLY',
      'NHID',
    ]),
    list([
      'NHIE',
      'NHIP',
      'NILE',
      'NLEU',
      'NLYS',
      'NME',
      'NMET',
      'NPHE',
    ]),
    list([
      'NPRO',
      'NSER',
      'NTHR',
      'NTRP',
      'NTYR',
      'NVAL',
      'ORN',
      'PGLU',
    ]),
    list([
      'PHE',
      'PRO',
      'QLN',
      'SER',
      'THR',
      'TRP',
      'TYR',
      'VAL',
    ]),
  ])
# ---
# name: test_StandardSelectionTable_protein_backbone
  list([
    list([
      'C',
      'CA',
      'N',
      'O',
    ]),
  ])
# ---
